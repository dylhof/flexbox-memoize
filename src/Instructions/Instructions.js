import React from 'react';

function Instructions(props) {
  return (
    <div>
      {props.showInstructions &&
      <div>
        <p>Instructions</p>
        <ol>
          <li>Click StartQuiz</li>
          <li>Choose the image that may be produced if the named flexbox property were in your CSS</li>
          <li>If you get stuck, check the sidebar to get more information about the flexbox property</li>
          <li>Enjoy creating simple layouts with flexbox!</li>
        </ol>
        <i class="far fa-times-circle" onClick={props.toggleShowInstructions}></i>
      </div>
      }
    </div>
  )
}

export default Instructions;