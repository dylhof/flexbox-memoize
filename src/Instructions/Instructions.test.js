import React from 'react';
import Instructions from './Instructions';
import { shallow } from 'enzyme';

const toggleShowInstructionsMock = jest.fn();

describe('Instructions', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Instructions 
        showInstructions={true}
        toggleShowInstructions={toggleShowInstructionsMock}/>
    )
  })
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})