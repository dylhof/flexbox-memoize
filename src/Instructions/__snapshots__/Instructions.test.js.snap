// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instructions should match the snapshot 1`] = `
<div>
  <div>
    <p>
      Instructions
    </p>
    <ol>
      <li>
        Click StartQuiz
      </li>
      <li>
        Choose the image that may be produced if the named flexbox property were in your CSS
      </li>
      <li>
        If you get stuck, check the sidebar to get more information about the flexbox property
      </li>
      <li>
        Enjoy creating simple layouts with flexbox!
      </li>
    </ol>
    <i
      class="far fa-times-circle"
      onClick={[MockFunction]}
    />
  </div>
</div>
`;
